sdk/core/AzureCore/Source/Pipeline/Policies/UserAgentPolicy.swift

```diff
@@ -57,24 +57,24 @@ public class UserAgentPolicy: PipelineStage {
 
     // MARK: Initializers
 
-    public convenience init(for clazz: AnyClass, applicationId: String? = nil) {
+    public convenience init(for clazz: AnyClass, telemetryOptions: TelemetryOptions) {
```

Comment:
`clazz`? How did I miss this before?

Reply:
Also, considering that the default of TelemetryOptions is all `nil`, why not make `telemetryOptions` optional here and simply instantiate a blank one if not specified? It would cut down on a lot of the other changes in this PR.

Reply:
Because `class` is typically a reserved keyword, "clazz" is a common name for such a parameter. However I could change it.

Reply:
I considered making TelemetryOptions optional, but I liked making it required because it contains the very important flag to disable telemetry, and I felt like setting the default value would be the job of the client. Honestly there's no strong reason to go either way.

Reply:
I'm thinking mainly from a maintainability standpoint of having to add boilerplate telemetryOptions every time a test is created.

But on that note, what is the policy on the default setting for telemetry being enabled? In CLI it was a whole thing because of privacy. @adrianhall @mayurid is this something we need to worry about?

Reply:
That's fair, I don't have a strong preference for this so I added a default value for the parameter in UserAgentPolicy.

Reply:
For the telemetry setting, I looked at the other SDKs and they have a flag to "disable" telemetry, which is false by default. (Rather than say, having a flag to "enable" telemetry which is true by default.) So this should match the semantics and naming that we use in our other language SDKs.

Reply:
I just recalled the specifics of why I did not make the TelemetryOptions parameter in the UserAgentPolicy initializer have a default value - it FORCES the clients to pass in a value when adding the policy to the pipeline, which will prevent the situation where a client accidentally forgets to pass the customer's TelemetryOptions and the default value is used instead (and since the default is that telemetry is enabled, we could accidentally be sending telemetry even if the customer disabled it which is VERY bad).

I added an extension to the test project which provides a convenience initializer that lets the tests omit the telemetryOptions parameter instead, this should resolve the concern while still maintaining the requirement for clients to always provide a TelemetryOptions.